src/layers/domain/core/reference.py

```diff
@@ -0,0 +1,30 @@
+class Reference:
```

Comment:
See comment above: basically - the reference class reads as a code smell when we've already introduced hashing in the entity class and objects are anyway treated as references in python